ndscan/plots/xy_1d.py

```diff
@@ -29,54 +44,119 @@ def __init__(self, view_box, data_name, data_item, error_bar_name, error_bar_ite
+            i1 = least_common_len
+            x_data, y_data, y_err = self._average_add_points(
+                x_data[i0:i1], y_data[i0:i1],
+                y_err[i0:i1] if self.error_bar_name else None)
+            num_to_show = len(x_data)
+        else:
+            num_to_show = least_common_len
+
         if self.plot_left_to_right:
             order = np.argsort(x_data[:num_to_show])
 
             self.data_item.setData(x_data[order], y_data[order])
-            if self.num_current_points == 0:
-                self.view_box.addItem(self.data_item)
 
-            if self.error_bar_item:
+            if self.error_bar_name or self._show_average_points:
                 self.error_bar_item.setData(x=x_data[order],
                                             y=y_data[order],
                                             height=y_err[order])
-                if self.num_current_points == 0:
-                    self.view_box.addItem(self.error_bar_item)
         else:
             self.data_item.setData(x_data[:num_to_show], y_data[:num_to_show])
-            if self.num_current_points == 0:
-                self.view_box.addItem(self.data_item)
 
-            if self.error_bar_item:
+            if self.error_bar_name or combine_points:
                 self.error_bar_item.setData(x=x_data[:num_to_show],
                                             y=y_data[:num_to_show],
                                             height=(2 * y_err[:num_to_show]))
-                if self.num_current_points == 0:
-                    self.view_box.addItem(self.error_bar_item)
 
-        self.num_current_points = num_to_show
+        if self.num_current_points == 0:
+            self.view_box.addItem(self.data_item)
+            if self.error_bar_name or self._show_average_points:
+                self.view_box.addItem(self.error_bar_item)
+
+        self.num_current_points = least_common_len
+
+    def _average_add_points(self, new_x_data, new_y_data, new_y_err=None):
+        # Append new data to collection.
+        if new_y_err is not None:
+            for (x, y, err) in zip(new_x_data, new_y_data, new_y_err):
+                self._y_collection[x].append(y)
+                self._err_collection[x].append(err)
+        else:
+            for (x, y) in zip(new_x_data, new_y_data):
+                self._y_collection[x].append(y)
+
+        # Average over values with same coordinate.
+        x_data = np.array(list(self._y_collection.keys()))
+        # Assuming all points contain the same number of samples.
+        y_data = np.array([np.mean(self._y_collection[x]) for x in x_data])
+
+        if new_y_err is not None:
+            y_err = np.array([
+                combined_uncertainty(self._y_collection[x], self._err_collection[x])
+                for x in x_data
+            ])
+        else:
+            y_err = np.array([
+                np.std(self._y_collection[x]) / np.sqrt(len(self._y_collection[x]))
```

Comment:
Shouldn't the divisor be `n - 1` here? (Possibly with some extra Student-t-weighting, even.